src/services/DataTools.js

```diff
@@ -0,0 +1,70 @@
+const ColorConsole = require('../services/ColorConsole');
+// const ConfigService = require('./services/Config/ConfigService');
+// TODO use ConfigService getTestRailStatusSkipped, getTestRailStatusFailed, 
+//    and getTestRailStatusPassed Instead of it 2, 5, 1
+
+const RESULT_STATUS_FAILED = 5;
+const RESULT_STATUS_SKIP = 2;
+// const RESULT_STATUS_PASS = 1;
+
+
+class DataTools {
+
+  
+  sayHiCyTr() {
+    ColorConsole.debug('Hi from DataTools.sayHiCyTr' );
+  }
+
+  /**
+   *
+   * @param {cytrPostData}
+   * @returns {*}
+   */
+  aggregateDuplicateResults(cytrPostData) {
+    // Replace multiple occurrences of test cases with one output summarizing case, aggregation rules are;
+    // Fail trumps all, anything with a failed leads to the first failed item is the one emitted
+    // If no mixed outcome, the same result is passed through
+    // Pass trumps skip
+    // If all test pass. The last comment is the one emitted.
+    // if all pass or skip modify the comment statement with the prefix of `Summarization of {n} Cypress tests` 
+    const emitPostData = []
+    const mergedResults = {} // this is a map of case_id to merged result object
+
+    cytrPostData.forEach((postData) => {
+      const { case_id, status_id, comment, elapsed, screenshotPaths } = postData;
+      if (Object.hasOwnProperty.call(mergedResults, case_id)) {
+        if (mergedResults[case_id].status_id !== RESULT_STATUS_FAILED) { // If one case failed don't process additional cases, Fail trumps all
+          mergedResults[case_id].count += 1;
+          if (status_id !== RESULT_STATUS_SKIP) { // If the current case is a skip then don't processing itFail trumps all, fail and pass Trump skip
+                    
+            mergedResults[case_id].status_id = status_id;
+            mergedResults[case_id].comment = comment;
+            mergedResults[case_id].elapsed = elapsed;
+            mergedResults[case_id].screenshotPaths = screenshotPaths;
```

Comment:
This is an array, do you also want to combine screenshots instead of the last case winning?
```suggestion
            mergedResults[case_id].screenshotPaths = mergedResults[case_id].screenshotPaths.concat(screenshotPaths);
```

Reply:
My design intentions are that if a failure occurs, the human is going to need to look into the matter and probably will go to Malcolm. Awesome. I don't look at the code directly or run the cypress app. Test rails is not a suitable vehicle to contain sufficient Space to contain everything needed to debug the problem directly. 

As such, I did not aggregate all of the failures together and make a complete failure report I simply collect the report from the first failure and it's one screenshot. Also, we've talked about this in several meetings and it is somewhat annoying to have it brought up nitty-gritty code review. I've tried to engage you on this topic three or four times. I thought I had your sign off on this methodology.